diff -uNr a/kernel/nvidia-modules-common.mk b/kernel/nvidia-modules-common.mk
--- a/kernel/nvidia-modules-common.mk	2013-11-19 19:01:49.345801177 +0100
+++ b/kernel/nvidia-modules-common.mk	2013-11-19 19:02:09.450801841 +0100
@@ -112,7 +112,7 @@
 # NVIDIA specific CFLAGS and #define's.
 #
 
-EXTRA_CFLAGS += -D__KERNEL__ -DMODULE -DNVRM -DNV_VERSION_STRING=\"331.20\" -Wno-unused-function -Wuninitialized -mno-red-zone -mcmodel=kernel -DNV_UVM_ENABLE -D__linux__ -DNV_DEV_NAME=\"$(MODULE_NAME)\"
+EXTRA_CFLAGS += -D__KERNEL__ -DMODULE -DNVRM -DNV_VERSION_STRING=\"331.20\" -Wno-unused-function -Wuninitialized -mno-red-zone -mcmodel=kernel -D__linux__ -DNV_DEV_NAME=\"$(MODULE_NAME)\"
 
 #
 # Detect SGI UV systems and apply system-specific optimizations.
