diff -uNr a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2014-01-16 19:39:46.789331910 +0100
+++ b/CMakeLists.txt	2014-01-16 19:40:23.984332501 +0100
@@ -159,9 +159,6 @@
 endif(WITH_TESTS)
 
 set(QT_REQUIRED_MODULES QtCore QtGui QtTest)
-if(UNIX AND NOT APPLE)
-  set(QT_REQUIRED_MODULES ${QT_REQUIRED_MODULES} QtDBus)
-endif()
 
 find_package(Qt4 4.6.0 REQUIRED ${QT_REQUIRED_MODULES})
 include(${QT_USE_FILE})
diff -uNr a/src/CMakeLists.txt b/src/CMakeLists.txt
--- a/src/CMakeLists.txt	2014-01-16 19:39:46.838331910 +0100
+++ b/src/CMakeLists.txt	2014-01-16 19:40:55.953333009 +0100
@@ -216,10 +216,6 @@
                       ${GCRYPT_LIBRARIES}
                       ${ZLIB_LIBRARIES})
 
-if(UNIX AND NOT APPLE)
-  target_link_libraries(${PROGNAME} ${QT_QTDBUS_LIBRARY})
-endif()
-
 set_target_properties(${PROGNAME} PROPERTIES ENABLE_EXPORTS ON)
 
 if(APPLE)
diff -uNr a/src/gui/Clipboard.cpp b/src/gui/Clipboard.cpp
--- a/src/gui/Clipboard.cpp	2014-01-16 19:39:46.838331910 +0100
+++ b/src/gui/Clipboard.cpp	2014-01-16 19:41:36.916333661 +0100
@@ -21,11 +21,6 @@
 #include <QClipboard>
 #include <QTimer>
 
-#if defined(Q_OS_UNIX) && !defined(Q_OS_MAC)
-#include <QDBusConnection>
-#include <QDBusMessage>
-#endif
-
 #include "core/Config.h"
 
 Clipboard* Clipboard::m_instance(Q_NULLPTR);
@@ -69,11 +64,6 @@
     if (clipboard->supportsSelection()) {
         clipboard->clear(QClipboard::Selection);
     }
-
-#if defined(Q_OS_UNIX) && !defined(Q_OS_MAC)
-    QDBusMessage message = QDBusMessage::createMethodCall("org.kde.klipper", "/klipper", "", "clearClipboardHistory");
-    QDBusConnection::sessionBus().send(message);
-#endif
 }
 
 void Clipboard::cleanup()
diff -uNr a/tests/CMakeLists.txt b/tests/CMakeLists.txt
--- a/tests/CMakeLists.txt	2014-01-16 19:39:46.773331909 +0100
+++ b/tests/CMakeLists.txt	2014-01-16 19:41:08.264333205 +0100
@@ -96,10 +96,6 @@
     ${ZLIB_LIBRARIES}
 )
 
-if(UNIX AND NOT APPLE)
-  set(TEST_LIBRARIES ${TEST_LIBRARIES} ${QT_QTDBUS_LIBRARY})
-endif()
-
 set(modeltest_SOURCRS modeltest.cpp)
 qt4_wrap_cpp(modeltest_SOURCRS modeltest.h)
 add_library(modeltest STATIC ${modeltest_SOURCRS})
