diff -uNr a/gtk/gtkmountoperation.c b/gtk/gtkmountoperation.c
--- a/gtk/gtkmountoperation.c	2017-01-07 13:52:36.784053116 +0100
+++ b/gtk/gtkmountoperation.c	2017-01-07 14:40:49.226208968 +0100
@@ -30,7 +30,9 @@
 #include "gtkmountoperationprivate.h"
 #include "gtkbox.h"
 #include "gtkcssiconthemevalueprivate.h"
+#if 0
 #include "gtkdbusgenerated.h"
+#endif
 #include "gtkentry.h"
 #include "gtkbox.h"
 #include "gtkintl.h"
@@ -54,7 +54,9 @@
 #include "gtksettings.h"
 #include "gtkstylecontextprivate.h"
 
+#if 0
 #include <glib/gprintf.h>
+#endif
 
 /**
  * SECTION:filesystem
@@ -113,9 +115,11 @@
   GdkScreen *screen;
 
   /* bus proxy */
+#if 0
   _GtkMountOperationHandler *handler;
   GCancellable *cancellable;
   gboolean handler_showing;
+#endif
 
   /* for the ask-password dialog */
   GtkWidget *grid;
@@ -189,6 +189,7 @@
 
   operation->priv = gtk_mount_operation_get_instance_private (operation);
 
+#if 0
   operation->priv->handler =
     _gtk_mount_operation_handler_proxy_new_for_bus_sync (G_BUS_TYPE_SESSION,
                                                          G_DBUS_PROXY_FLAGS_DO_NOT_AUTO_START,
@@ -202,6 +203,7 @@
 
   if (operation->priv->handler)
     g_dbus_proxy_set_default_timeout (G_DBUS_PROXY (operation->priv->handler), G_MAXINT);
+#endif
 }
 
 static void
@@ -224,8 +226,10 @@
   if (priv->screen)
     g_object_unref (priv->screen);
 
+#if 0
   if (priv->handler)
     g_object_unref (priv->handler);
+#endif
 
   G_OBJECT_CLASS (gtk_mount_operation_parent_class)->finalize (object);
 }
@@ -271,7 +275,11 @@
       break;
 
     case PROP_IS_SHOWING:
+#if 0
       g_value_set_boolean (value, priv->dialog != NULL || priv->handler_showing);
+#else
+      g_value_set_boolean (value, priv->dialog != NULL);
+#endif
       break;
 
     case PROP_SCREEN:
@@ -284,6 +292,7 @@
     }
 }
 
+#if 0
 static void
 gtk_mount_operation_proxy_finish (GtkMountOperation     *op,
                                   GMountOperationResult  result)
@@ -298,6 +307,7 @@
   /* drop the reference acquired when calling the proxy method */
   g_object_unref (op);
 }
+#endif
 
 static void
 remember_button_toggled (GtkToggleButton   *button,
@@ -739,6 +749,7 @@
   g_object_ref (operation);
 }
 
+#if 0
 static void
 call_password_proxy_cb (GObject      *source,
                         GAsyncResult *res,
@@ -799,6 +810,7 @@
                                                   operation->priv->ask_flags, NULL,
                                                   call_password_proxy_cb, operation);
 }
+#endif
 
 static void
 gtk_mount_operation_ask_password (GMountOperation   *mount_op,
@@ -815,6 +827,7 @@
   priv = operation->priv;
   priv->ask_flags = flags;
 
+#if 0
   use_gtk = (operation->priv->handler == NULL) ||
     (priv->ask_flags & G_ASK_PASSWORD_NEED_DOMAIN) ||
     (priv->ask_flags & G_ASK_PASSWORD_NEED_USERNAME);
@@ -823,6 +836,9 @@
     gtk_mount_operation_ask_password_do_gtk (operation, message, default_user, default_domain);
   else
     gtk_mount_operation_ask_password_do_proxy (operation, message, default_user, default_domain);
+#else
+  gtk_mount_operation_ask_password_do_gtk (operation, message, default_user, default_domain);
+#endif
 }
 
 static void
@@ -906,6 +922,7 @@
   g_object_ref (op);
 }
 
+#if 0
 static void
 call_question_proxy_cb (GObject      *source,
                         GAsyncResult *res,
@@ -962,6 +979,7 @@
                                                   choices, NULL,
                                                   call_question_proxy_cb, operation);
 }
+#endif
 
 static void
 gtk_mount_operation_ask_question (GMountOperation *op,
@@ -972,12 +990,17 @@
   gboolean use_gtk;
 
   operation = GTK_MOUNT_OPERATION (op);
+#if 0
   use_gtk = (operation->priv->handler == NULL);
 
   if (use_gtk)
     gtk_mount_operation_ask_question_do_gtk (operation, message, choices);
   else
     gtk_mount_operation_ask_question_do_proxy (operation, message, choices);
+#else
+  gtk_mount_operation_ask_question_do_gtk (operation, message, choices);
+#endif
+
 }
 
 static void
@@ -1505,6 +1528,7 @@
   return dialog;
 }
 
+#if 0
 static void
 call_processes_proxy_cb (GObject     *source,
                         GAsyncResult *res,
@@ -1571,6 +1595,7 @@
                                                     choices, NULL,
                                                     call_processes_proxy_cb, operation);
 }
+#endif
 
 static void
 gtk_mount_operation_show_processes_do_gtk (GtkMountOperation *op,
@@ -1618,12 +1643,16 @@
   gboolean use_gtk;
 
   operation = GTK_MOUNT_OPERATION (op);
+#if 0
   use_gtk = (operation->priv->handler == NULL);
 
   if (use_gtk)
     gtk_mount_operation_show_processes_do_gtk (operation, message, processes, choices);
   else
     gtk_mount_operation_show_processes_do_proxy (operation, message, processes, choices);
+#else
+  gtk_mount_operation_show_processes_do_gtk (operation, message, processes, choices);
+#endif
 }
 
 static void
@@ -1641,6 +1670,7 @@
       g_object_unref (op);
     }
 
+#if 0
   if (priv->handler != NULL)
     {
       _gtk_mount_operation_handler_call_close (priv->handler, NULL, NULL, NULL);
@@ -1648,6 +1678,7 @@
       priv->handler_showing = FALSE;
       g_object_notify (G_OBJECT (op), "is-showing");
     }
+#endif
 }
 
 /**
