diff -uNr a/configure b/configure
--- a/configure	2017-01-07 13:52:36.814053118 +0100
+++ b/configure	2017-01-07 13:53:05.475054662 +0100
@@ -26897,7 +26897,7 @@
 ########################################
 
 if test x$enable_x11_backend = xyes; then
-   ATK_PACKAGES="atk atk-bridge-2.0"
+   ATK_PACKAGES="atk"
 else
    ATK_PACKAGES="atk"
 fi
diff -uNr a/configure.ac b/configure.ac
--- a/configure.ac	2017-01-07 13:52:36.815053118 +0100
+++ b/configure.ac	2017-01-07 13:53:05.475054662 +0100
@@ -1371,7 +1371,7 @@
 ########################################
 
 if test x$enable_x11_backend = xyes; then
-   ATK_PACKAGES="atk atk-bridge-2.0"
+   ATK_PACKAGES="atk"
 else
    ATK_PACKAGES="atk"
 fi
diff -uNr a/gtk/a11y/gtkaccessibility.c b/gtk/a11y/gtkaccessibility.c
--- a/gtk/a11y/gtkaccessibility.c	2017-01-07 13:52:36.798053117 +0100
+++ b/gtk/a11y/gtkaccessibility.c	2017-01-07 13:53:05.225054649 +0100
@@ -37,7 +37,7 @@
 #include <gtk/gtktogglebutton.h>
 #include <gtk/gtkaccessible.h>
 
-#ifdef GDK_WINDOWING_X11
+#if 0
 #include <atk-bridge.h>
 #endif
 
@@ -988,7 +988,7 @@
   _gtk_accessibility_override_atk_util ();
   do_window_event_initialization ();
 
-#ifdef GDK_WINDOWING_X11
+#if 0
   atk_bridge_adaptor_init (NULL, NULL);
 #endif
 
