diff -uNr a/gtk/gtkapplicationimpl.c b/gtk/gtkapplicationimpl.c
--- a/gtk/gtkapplicationimpl.c	2015-03-02 20:34:03.000000000 +0100
+++ b/gtk/gtkapplicationimpl.c	2015-07-25 12:39:10.937176170 +0200
@@ -172,7 +172,7 @@
 
 #ifdef GDK_WINDOWING_X11
   if (GDK_IS_X11_DISPLAY (display))
-    impl_type = gtk_application_impl_x11_get_type ();
+    impl_type = gtk_application_impl_get_type ();
 #endif
 
 #ifdef GDK_WINDOWING_WAYLAND
diff -uNr a/gtk/gtkapplication-x11.c b/gtk/gtkapplication-x11.c
--- a/gtk/gtkapplication-x11.c	2015-03-02 20:34:03.000000000 +0100
+++ b/gtk/gtkapplication-x11.c	2015-07-25 12:35:33.816172718 +0200
@@ -28,39 +28,25 @@
 
 typedef struct
 {
-  GtkApplicationImplDBus dbus;
-
 } GtkApplicationImplX11;
 
-G_DEFINE_TYPE (GtkApplicationImplX11, gtk_application_impl_x11, GTK_TYPE_APPLICATION_IMPL_DBUS)
+//G_DEFINE_TYPE (GtkApplicationImplX11, gtk_application_impl_x11, GTK_TYPE_APPLICATION_IMPL_DBUS)
 
 static void
 gtk_application_impl_x11_handle_window_realize (GtkApplicationImpl *impl,
                                                 GtkWindow          *window)
 {
-  GtkApplicationImplDBus *dbus = (GtkApplicationImplDBus *) impl;
   GdkWindow *gdk_window;
-  gchar *window_path;
 
   gdk_window = gtk_widget_get_window (GTK_WIDGET (window));
 
   if (!GDK_IS_X11_WINDOW (gdk_window))
     return;
 
-  window_path = gtk_application_impl_dbus_get_window_path (dbus, window);
-
-  gdk_x11_window_set_utf8_property (gdk_window, "_GTK_APPLICATION_ID", dbus->application_id);
-  gdk_x11_window_set_utf8_property (gdk_window, "_GTK_UNIQUE_BUS_NAME", dbus->unique_name);
-  gdk_x11_window_set_utf8_property (gdk_window, "_GTK_APPLICATION_OBJECT_PATH", dbus->object_path);
-  gdk_x11_window_set_utf8_property (gdk_window, "_GTK_WINDOW_OBJECT_PATH", window_path);
-  gdk_x11_window_set_utf8_property (gdk_window, "_GTK_APP_MENU_OBJECT_PATH", dbus->app_menu_path);
-  gdk_x11_window_set_utf8_property (gdk_window, "_GTK_MENUBAR_OBJECT_PATH", dbus->menubar_path);
-
-  g_free (window_path);
 }
 
 static GVariant *
-gtk_application_impl_x11_get_window_system_id (GtkApplicationImplDBus *dbus,
+gtk_application_impl_x11_get_window_system_id (
                                                GtkWindow              *window)
 {
   GdkWindow *gdk_window;
@@ -70,7 +56,6 @@
   if (GDK_IS_X11_WINDOW (gdk_window))
     return g_variant_new_uint32 (GDK_WINDOW_XID (gdk_window));
 
-  return GTK_APPLICATION_IMPL_DBUS_CLASS (gtk_application_impl_x11_parent_class)->get_window_system_id (dbus, window);
 }
 
 static void
@@ -95,10 +80,8 @@
 static void
 gtk_application_impl_x11_class_init (GtkApplicationImplX11Class *class)
 {
-  GtkApplicationImplDBusClass *dbus_class = GTK_APPLICATION_IMPL_DBUS_CLASS (class);
   GtkApplicationImplClass *impl_class = GTK_APPLICATION_IMPL_CLASS (class);
 
   impl_class->handle_window_realize = gtk_application_impl_x11_handle_window_realize;
-  dbus_class->get_window_system_id = gtk_application_impl_x11_get_window_system_id;
   impl_class->before_emit = gtk_application_impl_x11_before_emit;
 }
diff -uNr a/gtk/Makefile.in b/gtk/Makefile.in
--- a/gtk/Makefile.in	2015-07-25 12:58:42.755194800 +0200
+++ b/gtk/Makefile.in	2015-07-25 12:20:47.008158619 +0200
@@ -372,7 +372,7 @@
 	gtkprintoperation-win32.c gtkwin32.c gtkplug.c gtksocket.c \
 	gtkxembed.c deprecated/gtktrayicon-x11.c gtkapplication-x11.c \
 	gtkmountoperation-x11.c gtkapplication-wayland.c \
-	gtkapplication-dbus.c gtkwin32embed.c gtkwin32embedwidget.c \
+	gtkwin32embed.c gtkwin32embedwidget.c \
 	gtkmountoperation-stub.c gtksearchenginequartz.c \
 	gtkapplication-quartz.c gtkapplication-quartz-menu.c \
 	gtkquartz.c
@@ -733,7 +733,7 @@
 	libgtk_3_la-gtkmountoperation-x11.lo
 @USE_X11_TRUE@am__objects_12 = $(am__objects_11)
 am__objects_13 = libgtk_3_la-gtkapplication-wayland.lo
-am__objects_14 = libgtk_3_la-gtkapplication-dbus.lo
+am__objects_14 = 
 @USE_WAYLAND_TRUE@am__objects_15 = $(am__objects_13) $(am__objects_14)
 @USE_WAYLAND_FALSE@@USE_X11_TRUE@am__objects_16 = $(am__objects_14)
 am__objects_17 = libgtk_3_la-gtkwin32embed.lo \
@@ -752,7 +752,7 @@
 	$(am__objects_12) $(am__objects_15) $(am__objects_16) \
 	$(am__objects_18) $(am__objects_20) $(am__objects_22)
 am_libgtk_3_la_OBJECTS = $(am__objects_23)
-am__objects_24 = libgtk_3_la-gtkdbusgenerated.lo
+am__objects_24 = 
 am__objects_25 = $(am__objects_24)
 nodist_libgtk_3_la_OBJECTS = $(am__objects_25)
 libgtk_3_la_OBJECTS = $(am_libgtk_3_la_OBJECTS) \
@@ -1134,7 +1134,7 @@
 	inspector/Makefile.inc libgtk3.manifest.in gtk-win32.rc.in \
 	gtkwin32embed.h gtkwin32embedwidget.h gtkwin32embedwidget.c \
 	gtkprint-win32.h gtkprint-win32.c gtksearchenginequartz.h \
-	gtkdbusinterfaces.xml gtk-keys.css.default gtk-keys.css.emacs \
+	gtk-keys.css.default gtk-keys.css.emacs \
 	gtk-keys.css.mac makefile.msc makefile.msc.in \
 	gtktypebuiltins.c.template gtktypebuiltins.h.template \
 	gtkprivatetypebuiltins.c.template \
@@ -1928,11 +1928,10 @@
 	gtkviewport.c gtkwidget.c gtkwidgetpath.c gtkwindow.c \
 	gtkwindowgroup.c gtkwin32theme.c $(am__append_3) \
 	$(am__append_4)
-nodist_gtk_c_sources = $(gtk_dbus_built_sources)
+nodist_gtk_c_sources = 
 gtk_c_sources = $(gtk_base_c_sources) $(am__append_6) $(am__append_8) \
 	$(am__append_9) $(am__append_11) $(am__append_12) \
 	$(am__append_13) $(am__append_15) $(am__append_17)
-gtk_dbus_built_sources = gtkdbusgenerated.c gtkdbusgenerated.h
 gtk_os_unix_c_sources = \
 	gtkcustompaperunixdialog.c	\
 	gtkpagesetupunixdialog.c	\
@@ -1962,8 +1961,7 @@
 gtk_use_wayland_c_sources = \
 	gtkapplication-wayland.c
 
-gtk_use_wayland_or_x11_c_sources = \
-	gtkapplication-dbus.c
+gtk_use_wayland_or_x11_c_sources = 
 
 gtk_use_win32_c_sources = \
 	gtkwin32embed.c		\
@@ -2054,7 +2052,6 @@
 #
 MAINTAINERCLEANFILES = \
 	$(gtk_built_sources) 			\
-	$(gtk_dbus_built_sources)		\
 	$(stamp_files)
 
 DISTCLEANFILES = 
@@ -2096,8 +2093,7 @@
 gen_sources = xgen-gdef xgen-gtbh xgen-gtic xgen-gmh xgen-gmc xgen-gmlh xgen-gmlc xgen-gtfsrc.c xgen-gtf
 CLEANFILES = $(gen_sources) $(am__append_24)
 BUILT_SOURCES = \
-	$(gtk_built_sources) \
-	$(gtk_dbus_built_sources)
+	$(gtk_built_sources)
 
 theme_sources = \
 	theme/win32/gtk-win32.css 		\
@@ -2659,7 +2655,6 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libgtk_3_la-gtkappchooserbutton.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libgtk_3_la-gtkappchooserdialog.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libgtk_3_la-gtkappchooserwidget.Plo@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libgtk_3_la-gtkapplication-dbus.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libgtk_3_la-gtkapplication-quartz-menu.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libgtk_3_la-gtkapplication-quartz.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libgtk_3_la-gtkapplication-wayland.Plo@am__quote@
@@ -2768,7 +2763,6 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libgtk_3_la-gtkcssunsetvalue.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libgtk_3_la-gtkcssvalue.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libgtk_3_la-gtkcustompaperunixdialog.Plo@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libgtk_3_la-gtkdbusgenerated.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libgtk_3_la-gtkdialog.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libgtk_3_la-gtkdnd-quartz.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libgtk_3_la-gtkdnd.Plo@am__quote@
@@ -6404,13 +6398,6 @@
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
 @am__fastdepCC_FALSE@	$(AM_V_CC@am__nodep@)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libgtk_3_la_CFLAGS) $(CFLAGS) -c -o libgtk_3_la-gtkapplication-wayland.lo `test -f 'gtkapplication-wayland.c' || echo '$(srcdir)/'`gtkapplication-wayland.c
 
-libgtk_3_la-gtkapplication-dbus.lo: gtkapplication-dbus.c
-@am__fastdepCC_TRUE@	$(AM_V_CC)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libgtk_3_la_CFLAGS) $(CFLAGS) -MT libgtk_3_la-gtkapplication-dbus.lo -MD -MP -MF $(DEPDIR)/libgtk_3_la-gtkapplication-dbus.Tpo -c -o libgtk_3_la-gtkapplication-dbus.lo `test -f 'gtkapplication-dbus.c' || echo '$(srcdir)/'`gtkapplication-dbus.c
-@am__fastdepCC_TRUE@	$(AM_V_at)$(am__mv) $(DEPDIR)/libgtk_3_la-gtkapplication-dbus.Tpo $(DEPDIR)/libgtk_3_la-gtkapplication-dbus.Plo
-@AMDEP_TRUE@@am__fastdepCC_FALSE@	$(AM_V_CC)source='gtkapplication-dbus.c' object='libgtk_3_la-gtkapplication-dbus.lo' libtool=yes @AMDEPBACKSLASH@
-@AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
-@am__fastdepCC_FALSE@	$(AM_V_CC@am__nodep@)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libgtk_3_la_CFLAGS) $(CFLAGS) -c -o libgtk_3_la-gtkapplication-dbus.lo `test -f 'gtkapplication-dbus.c' || echo '$(srcdir)/'`gtkapplication-dbus.c
-
 libgtk_3_la-gtkwin32embed.lo: gtkwin32embed.c
 @am__fastdepCC_TRUE@	$(AM_V_CC)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libgtk_3_la_CFLAGS) $(CFLAGS) -MT libgtk_3_la-gtkwin32embed.lo -MD -MP -MF $(DEPDIR)/libgtk_3_la-gtkwin32embed.Tpo -c -o libgtk_3_la-gtkwin32embed.lo `test -f 'gtkwin32embed.c' || echo '$(srcdir)/'`gtkwin32embed.c
 @am__fastdepCC_TRUE@	$(AM_V_at)$(am__mv) $(DEPDIR)/libgtk_3_la-gtkwin32embed.Tpo $(DEPDIR)/libgtk_3_la-gtkwin32embed.Plo
@@ -6460,13 +6447,6 @@
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
 @am__fastdepCC_FALSE@	$(AM_V_CC@am__nodep@)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libgtk_3_la_CFLAGS) $(CFLAGS) -c -o libgtk_3_la-gtkquartz.lo `test -f 'gtkquartz.c' || echo '$(srcdir)/'`gtkquartz.c
 
-libgtk_3_la-gtkdbusgenerated.lo: gtkdbusgenerated.c
-@am__fastdepCC_TRUE@	$(AM_V_CC)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libgtk_3_la_CFLAGS) $(CFLAGS) -MT libgtk_3_la-gtkdbusgenerated.lo -MD -MP -MF $(DEPDIR)/libgtk_3_la-gtkdbusgenerated.Tpo -c -o libgtk_3_la-gtkdbusgenerated.lo `test -f 'gtkdbusgenerated.c' || echo '$(srcdir)/'`gtkdbusgenerated.c
-@am__fastdepCC_TRUE@	$(AM_V_at)$(am__mv) $(DEPDIR)/libgtk_3_la-gtkdbusgenerated.Tpo $(DEPDIR)/libgtk_3_la-gtkdbusgenerated.Plo
-@AMDEP_TRUE@@am__fastdepCC_FALSE@	$(AM_V_CC)source='gtkdbusgenerated.c' object='libgtk_3_la-gtkdbusgenerated.lo' libtool=yes @AMDEPBACKSLASH@
-@AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
-@am__fastdepCC_FALSE@	$(AM_V_CC@am__nodep@)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libgtk_3_la_CFLAGS) $(CFLAGS) -c -o libgtk_3_la-gtkdbusgenerated.lo `test -f 'gtkdbusgenerated.c' || echo '$(srcdir)/'`gtkdbusgenerated.c
-
 mostlyclean-libtool:
 	-rm -f *.lo
 
@@ -6960,13 +6940,6 @@
 
 .PHONY: included-modules
 
-$(gtk_dbus_built_sources) : Makefile.am gtkdbusinterfaces.xml
-	$(AM_V_GEN) gdbus-codegen \
-	--interface-prefix org.Gtk. \
-	--c-namespace _Gtk \
-	--generate-c-code gtkdbusgenerated \
-	$(srcdir)/gtkdbusinterfaces.xml
-
 # all autogenerated files need to be generated in the srcdir,
 # so old versions get remade and are not confused with newer
 # versions in the build dir. thus a development setup requires
@@ -7092,7 +7065,7 @@
 dist-hook: ../build/win32/vs9/gtk.vcproj ../build/win32/vs10/gtk.vcxproj ../build/win32/vs10/gtk.vcxproj.filters
 
 ../build/win32/vs9/gtk.vcproj: ../build/win32/vs9/gtk.vcprojin
-	for F in `echo $(gtk_base_c_sources) $(gtk_os_win32_c_sources) $(gtk_use_win32_c_sources) $(gtk_dbus_built_sources) | tr '/' '\\'`; do \
+	for F in `echo $(gtk_base_c_sources) $(gtk_os_win32_c_sources) $(gtk_use_win32_c_sources) | tr '/' '\\'`; do \
 		case $$F in \
 		*.c) echo '   <File RelativePath="..\..\..\gtk\'$$F'" />' \
 		     ;; \
@@ -7102,7 +7075,7 @@
 	rm libgtk.sourcefiles
 
 ../build/win32/vs10/gtk.vcxproj: ../build/win32/vs10/gtk.vcxprojin
-	for F in `echo $(gtk_base_c_sources) $(gtk_os_win32_c_sources) $(gtk_use_win32_c_sources) $(gtk_dbus_built_sources) | tr '/' '\\'`; do \
+	for F in `echo $(gtk_base_c_sources) $(gtk_os_win32_c_sources) $(gtk_use_win32_c_sources) | tr '/' '\\'`; do \
 		case $$F in \
 		*.c) echo '   <ClCompile Include="..\..\..\gtk\'$$F'" />' \
 		     ;; \
@@ -7112,7 +7085,7 @@
 	rm libgtk.vs10.sourcefiles
 
 ../build/win32/vs10/gtk.vcxproj.filters: ../build/win32/vs10/gtk.vcxproj.filtersin
-	for F in `echo $(gtk_base_c_sources) $(gtk_os_win32_c_sources) $(gtk_use_win32_c_sources) $(gtk_dbus_built_sources) | tr '/' '\\'`; do \
+	for F in `echo $(gtk_base_c_sources) $(gtk_os_win32_c_sources) $(gtk_use_win32_c_sources) | tr '/' '\\'`; do \
 		case $$F in \
 		*.c) echo '   <ClCompile Include="..\..\..\gtk\'$$F'"><Filter>Source Files</Filter></ClCompile>' \
 		     ;; \
