diff -uNr a/gtk/gtkmountoperation.c b/gtk/gtkmountoperation.c
--- a/gtk/gtkmountoperation.c	2014-02-03 20:09:12.429384274 +0100
+++ b/gtk/gtkmountoperation.c	2014-02-03 20:09:27.844384519 +0100
@@ -29,7 +29,9 @@
 
 #include "gtkmountoperationprivate.h"
 #include "gtkbox.h"
+#if 0
 #include "gtkdbusgenerated.h"
+#endif
 #include "gtkentry.h"
 #include "gtkbox.h"
 #include "gtkintl.h"
@@ -49,7 +51,9 @@
 #include "gtkmenuitem.h"
 #include "gtkmain.h"
 
+#if 0
 #include <glib/gprintf.h>
+#endif
 
 /**
  * SECTION:filesystem
@@ -107,10 +111,12 @@
   GtkDialog *dialog;
   GdkScreen *screen;
 
+#if 0
   /* bus proxy */
   _GtkMountOperationHandler *handler;
   GCancellable *cancellable;
   gboolean handler_showing;
+#endif
 
   /* for the ask-password dialog */
   GtkWidget *grid;
@@ -182,10 +188,12 @@
 static void
 gtk_mount_operation_init (GtkMountOperation *operation)
 {
+#if 0
   gchar *name_owner;
+#endif
 
   operation->priv = gtk_mount_operation_get_instance_private (operation);
-
+#if 0
   operation->priv->handler =
     _gtk_mount_operation_handler_proxy_new_for_bus_sync (G_BUS_TYPE_SESSION,
                                                          G_DBUS_PROXY_FLAGS_DO_NOT_AUTO_START,
@@ -199,6 +207,7 @@
 
   if (operation->priv->handler)
     g_dbus_proxy_set_default_timeout (G_DBUS_PROXY (operation->priv->handler), G_MAXINT);
+#endif
 }
 
 static void
@@ -216,8 +225,10 @@
   if (priv->screen)
     g_object_unref (priv->screen);
 
+#if 0
   if (priv->handler)
     g_object_unref (priv->handler);
+#endif
 
   G_OBJECT_CLASS (gtk_mount_operation_parent_class)->finalize (object);
 }
@@ -263,7 +274,11 @@
       break;
 
     case PROP_IS_SHOWING:
+#if 0
       g_value_set_boolean (value, priv->dialog != NULL || priv->handler_showing);
+#else
+      g_value_set_boolean (value, priv->dialog != NULL);
+#endif
       break;
 
     case PROP_SCREEN:
@@ -276,6 +291,7 @@
     }
 }
 
+#if 0
 static void
 gtk_mount_operation_proxy_finish (GtkMountOperation     *op,
                                   GMountOperationResult  result)
@@ -290,6 +306,7 @@
   /* drop the reference acquired when calling the proxy method */
   g_object_unref (op);
 }
+#endif
 
 static void
 remember_button_toggled (GtkToggleButton   *button,
@@ -721,6 +738,7 @@
   g_object_ref (operation);
 }
 
+#if 0
 static void
 call_password_proxy_cb (GObject      *source,
                         GAsyncResult *res,
@@ -781,6 +799,7 @@
                                                   operation->priv->ask_flags, NULL,
                                                   call_password_proxy_cb, operation);
 }
+#endif
 
 static void
 gtk_mount_operation_ask_password (GMountOperation   *mount_op,
@@ -797,6 +816,7 @@
   priv = operation->priv;
   priv->ask_flags = flags;
 
+#if 0
   use_gtk = (operation->priv->handler == NULL) ||
     (priv->ask_flags & G_ASK_PASSWORD_NEED_DOMAIN) ||
     (priv->ask_flags & G_ASK_PASSWORD_NEED_USERNAME);
@@ -805,6 +825,9 @@
     gtk_mount_operation_ask_password_do_gtk (operation, message, default_user, default_domain);
   else
     gtk_mount_operation_ask_password_do_proxy (operation, message, default_user, default_domain);
+#else
+    gtk_mount_operation_ask_password_do_gtk (operation, message, default_user, default_domain);
+#endif
 }
 
 static void
@@ -888,6 +911,7 @@
   g_object_ref (op);
 }
 
+#if 0
 static void
 call_question_proxy_cb (GObject      *source,
                         GAsyncResult *res,
@@ -944,6 +968,7 @@
                                                   choices, NULL,
                                                   call_question_proxy_cb, operation);
 }
+#endif
 
 static void
 gtk_mount_operation_ask_question (GMountOperation *op,
@@ -954,12 +979,16 @@
   gboolean use_gtk;
 
   operation = GTK_MOUNT_OPERATION (op);
+#if 0
   use_gtk = (operation->priv->handler == NULL);
 
   if (use_gtk)
     gtk_mount_operation_ask_question_do_gtk (operation, message, choices);
   else
     gtk_mount_operation_ask_question_do_proxy (operation, message, choices);
+#else
+    gtk_mount_operation_ask_question_do_gtk (operation, message, choices);
+#endif
 }
 
 static void
@@ -1505,6 +1534,7 @@
   return dialog;
 }
 
+#if 0
 static void
 call_processes_proxy_cb (GObject     *source,
                         GAsyncResult *res,
@@ -1571,6 +1601,7 @@
                                                     choices, NULL,
                                                     call_processes_proxy_cb, operation);
 }
+#endif
 
 static void
 gtk_mount_operation_show_processes_do_gtk (GtkMountOperation *op,
@@ -1618,12 +1649,16 @@
   gboolean use_gtk;
 
   operation = GTK_MOUNT_OPERATION (op);
+#if 0
   use_gtk = (operation->priv->handler == NULL);
 
   if (use_gtk)
     gtk_mount_operation_show_processes_do_gtk (operation, message, processes, choices);
   else
     gtk_mount_operation_show_processes_do_proxy (operation, message, processes, choices);
+#else
+    gtk_mount_operation_show_processes_do_gtk (operation, message, processes, choices);
+#endif
 }
 
 static void
@@ -1641,6 +1676,7 @@
       g_object_unref (op);
     }
 
+#if 0
   if (priv->handler != NULL)
     {
       _gtk_mount_operation_handler_call_close (priv->handler, NULL, NULL, NULL);
@@ -1648,6 +1684,7 @@
       priv->handler_showing = FALSE;
       g_object_notify (G_OBJECT (op), "is-showing");
     }
+#endif
 }
 
 /**
diff -uNr a/gtk/Makefile.am b/gtk/Makefile.am
--- a/gtk/Makefile.am	2014-02-03 20:09:12.754384279 +0100
+++ b/gtk/Makefile.am	2014-02-03 20:09:27.847384519 +0100
@@ -908,18 +908,6 @@
 gtk_c_sources = $(gtk_base_c_sources)
 gtk_all_c_sources = $(gtk_base_c_sources)
 
-gtk_dbus_built_sources = gtkdbusgenerated.c gtkdbusgenerated.h
-
-$(gtk_dbus_built_sources) : Makefile.am gtkdbusinterfaces.xml
-	$(AM_V_GEN) gdbus-codegen \
-	--interface-prefix org.Gtk. \
-	--c-namespace _Gtk \
-	--generate-c-code gtkdbusgenerated \
-	$(srcdir)/gtkdbusinterfaces.xml
-
-nodist_gtk_c_sources += $(gtk_dbus_built_sources)
-gtk_all_c_sources += $(gtk_dbus_built_sources)
-
 gtk_os_unix_c_sources = 		\
 	gtkcustompaperunixdialog.c	\
 	gtkpagesetupunixdialog.c	\
@@ -1080,7 +1068,6 @@
 #
 MAINTAINERCLEANFILES = \
 	$(gtk_built_sources) 			\
-	$(gtk_dbus_built_sources)		\
 	$(stamp_files) 				\
 	$(GENERATED_ICONS) 			\
 	stock-icons/icon-theme.cache
@@ -1145,7 +1132,6 @@
 
 BUILT_SOURCES = \
 	$(gtk_built_sources) \
-	$(gtk_dbus_built_sources) \
 	stamp-icons
 
 # all autogenerated files need to be generated in the srcdir,
@@ -1256,7 +1242,7 @@
 dist-hook: ../build/win32/vs9/gtk.vcproj ../build/win32/vs10/gtk.vcxproj ../build/win32/vs10/gtk.vcxproj.filters
 
 ../build/win32/vs9/gtk.vcproj: ../build/win32/vs9/gtk.vcprojin
-	for F in `echo $(gtk_base_c_sources) $(gtk_os_win32_c_sources) $(gtk_use_win32_c_sources) $(gtk_dbus_built_sources) | tr '/' '\\'`; do \
+	for F in `echo $(gtk_base_c_sources) $(gtk_os_win32_c_sources) $(gtk_use_win32_c_sources) | tr '/' '\\'`; do \
 		case $$F in \
 		*.c) echo '   <File RelativePath="..\..\..\gtk\'$$F'" />' \
 		     ;; \
@@ -1266,7 +1252,7 @@
 	rm libgtk.sourcefiles
 
 ../build/win32/vs10/gtk.vcxproj: ../build/win32/vs10/gtk.vcxprojin
-	for F in `echo $(gtk_base_c_sources) $(gtk_os_win32_c_sources) $(gtk_use_win32_c_sources) $(gtk_dbus_built_sources) | tr '/' '\\'`; do \
+	for F in `echo $(gtk_base_c_sources) $(gtk_os_win32_c_sources) $(gtk_use_win32_c_sources) | tr '/' '\\'`; do \
 		case $$F in \
 		*.c) echo '   <ClCompile Include="..\..\..\gtk\'$$F'" />' \
 		     ;; \
@@ -1276,7 +1262,7 @@
 	rm libgtk.vs10.sourcefiles
 
 ../build/win32/vs10/gtk.vcxproj.filters: ../build/win32/vs10/gtk.vcxproj.filtersin
-	for F in `echo $(gtk_base_c_sources) $(gtk_os_win32_c_sources) $(gtk_use_win32_c_sources) $(gtk_dbus_built_sources) | tr '/' '\\'`; do \
+	for F in `echo $(gtk_base_c_sources) $(gtk_os_win32_c_sources) $(gtk_use_win32_c_sources) | tr '/' '\\'`; do \
 		case $$F in \
 		*.c) echo '   <ClCompile Include="..\..\..\gtk\'$$F'"><Filter>Source Files</Filter></ClCompile>' \
 		     ;; \
@@ -1769,7 +1755,6 @@
 	gtkprint-win32.c	\
 	gtksearchenginequartz.h \
 	gtk.gresource.xml	\
-	gtkdbusinterfaces.xml	\
 	gtk-default.css		\
 	gtk-keys.css.default	\
 	gtk-keys.css.emacs	\
